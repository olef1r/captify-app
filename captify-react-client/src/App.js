import React from 'react';
import Header from './components/layouts/Header';
import Footer from './components/layouts/Footer';
import SimpleTable from './components/Table';
import Button from './components/Button';
import TextField from './components/TextField';
import { postItem, getItems } from './services/bff';
import xss from 'xss';
import { isEmpty } from 'lodash'

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      value: '',
      data: [], 
      update: false
    };
  };

  componentDidMount() { 
    getItems().then(result => this.setState({ data: result.data }));
  };

  componentDidUpdate() {
    if (this.state.update) {
      getItems().then(result => this.setState({ data: result.data }));
      this.setState({update: !this.state.update})
    }
  }

  _onChange = e => this.setState({ value: e.target.value });
 
  _onClick = e => {
    e.preventDefault();
    const filteredData = xss(this.state.value);
    if (isEmpty(filteredData)) {
      return;
    }
    postItem({ item: filteredData }).then(() => {
      this.setState({ update: true, value: '' });
    });    
  }
   
  render() {   
    const isLimitExceeded = this.state.data.length < 10 ? false : true;
    return(
      <div>
        <Header />
        <SimpleTable rows={this.state.data} />
        <div style={{display: "inline-flex"}}>
          <TextField value={this.state.value} onChangeValue={this._onChange} />
          <Button onClick={this._onClick} disabled={isLimitExceeded} isValueEmpty/>
        </div>
        {isLimitExceeded  ? <p style={{display: "inline"}}>You cannot add more items</p> : null}
        <Footer />
      </div>
    );
  }
}

export default App;