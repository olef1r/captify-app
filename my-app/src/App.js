import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';


import Header from './components/layouts/Header';
import Footer from './components/layouts/Footer';
import SimpleTable from './components/Table';
import Button from './components/Button';
import  TextField from './components/TextField';
import { valueSelector } from './selectors/value.selectors';
import { loadValue } from './actions/value-action'


function App({value, loadValue}) {
  console.log("v", value)
  return (
    <div className="App">
      <Header />
      <SimpleTable rows={rows} />

        <TextField onChange = {loadValue}/>
        <Button/ >
     
      <Footer />
    </div>
  );
}

const mapStateToProps = state => ({
  value: valueSelector(state)
});

const mapDispatchToProps = dispatch => bindActionCreators({ loadValue }, dispatch);

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(App);
