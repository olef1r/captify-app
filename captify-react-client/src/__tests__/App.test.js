import React from 'react';
import App from '../App';

import { shallow } from 'enzyme';


let wrapper;

beforeEach(() => {
  wrapper = shallow(<App />);
});
describe('<App> - Component', () => { 

  it('renders without crashing', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper).toHaveDisplayName('div');
  });

  it('renders components', () => {
    expect(wrapper).toContainExactlyOneMatchingElement('OutlinedButtons');
    expect(wrapper).toContainExactlyOneMatchingElement('SimpleTable');
    expect(wrapper).toContainExactlyOneMatchingElement('BasicTextFields');
  });
});
