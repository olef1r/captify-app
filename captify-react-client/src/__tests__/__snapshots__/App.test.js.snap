// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App> - Component renders without crashing 1`] = `
<div>
  <Blog />
  <SimpleTable
    rows={Array []}
  />
  <div
    style={
      Object {
        "display": "inline-flex",
      }
    }
  >
    <BasicTextFields
      onChangeValue={[Function]}
      value=""
    />
    <OutlinedButtons
      disabled={false}
      isValueEmpty={true}
      onClick={[Function]}
    />
  </div>
  <Blog />
</div>
`;
